temporalcli/commands.workflow_view.go

```diff
@@ -265,6 +265,51 @@ func (*TemporalWorkflowCountCommand) run(*CommandContext, []string) error {
 	return fmt.Errorf("TODO")
 }
 
-func (*TemporalWorkflowShowCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowShowCommand) run(cctx *CommandContext, args []string) error {
+	// Call describe
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	// Print history
+	iter := &structuredHistoryIter{
+		ctx:            cctx,
+		client:         cl,
+		workflowID:     c.WorkflowId,
+		runID:          c.RunId,
+		includeDetails: true,
+		follow:         c.Follow,
+		color:          !cctx.JSONOutput && !color.NoColor,
+	}
+	if !cctx.JSONOutput {
+		cctx.Printer.Println(color.MagentaString("Progress:"))
+		// TODO: Do we need to do the "time elapsed" thing? Also needs some redoing to print
+		//   result at the end, since final event is not accessible from here.
+		if err := iter.print(cctx.Printer); err != nil {
+			return fmt.Errorf("displaying history failed: %w", err)
+		}
+	} else {
```

Comment:
It is important to note that IIUC `workflow show` is the primary way people dump workflow history for use in replayers. So for JSON, we need to dump exact history format (i.e. create the `history.History` proto object with its events and send that to `PrintStructured`, which is close to what's here but has to be the full thing)

Reply:
This is exactly how the old CLI did it FYI. But, making it be a literal history proto makes more sense anyway.

Reply:
> But, making it be a literal history proto makes more sense anyway.

They used that in `printReplayableHistory` which makes sense (it's simple code just for JSON), but for the non-JSON, yeah the lazy stuff for text is a bit more difficult. Feel free to steal anything from `workflow execute` which does similar.